core/integration/env.go

```diff
@@ -31,7 +31,6 @@ import (
 
```

Comment:
The separation of `impl/integration` from `core/integration` isn't documented, so let me explain here:
The layout of the code is designed to make using Key Transparency in an enterprise environment easy. Everything that might be implementation specific is in the `impl` directory.  In order for the `core` directory to operate independently of implementation details, it must not have any references to `impl`.  This is why the unit tests that rely only on the interfaces can remain in `core`, while the actual test that supplies the dependencies lives in `impl`. 

Please keep `impl/integration/env.go` in `impl`.

Reply:
I moved it there because of the cycle dependecies, so it cannot be solved without moving something, or env.go or all the tests to impl/integration.

Reply:
What was the cycle in particular?

Reply:
`cmd/gen-test-vectors` has call to `impl/integration`
and `impl/integration` has call to `integration/client_tests.go`
if we merge `cmd/gen-test-vectors` into `integration/client_tests.go` we get cycle.

Reply:
I think the link between `cmd/gen-test-vectors` to `impl/integration` can be split:
1. `main()` in `cmd/gen-test-vectors` is now `impl/integration/main_test.go`
1. `GenerateTestVectors` used to depend on `impl/integration/env.go`, but this can be swapped with `core/integration/env.go` 

There are a few other references to `impl` in `GenerateTestVectors` which I've annotated with fixes for.

Reply:
Ok, i'l try to do it in this way.